nixos/lib/test-driver/test-driver.py

```diff
@@ -958,6 +949,37 @@ def subtest(name: str) -> Iterator[None]:
     return False
 
 
+def _test_symbols() -> Dict[str, Any]:
+    general_symbols = dict(
+        os=os,
+        log=globals().get("log"),  # extracting those symbol keys
+        machines=globals().get("machines"),  # without being initialized
+        start_all=start_all,
+        create_machine=create_machine,
+        # run_tests=run_tests,
+        # join_all=join_all,
```

Comment:
`rg` in the whole of nixpkgs didn't yield any usage.

Reply:
I think the `join_all` method is valuable and it doesn't hurt to continue to expose it. So, i would be more in favor of exposing it. (However, if we don't expose `join_all`, would it be still useful to keep this method? IIUC, it could not be used.)
Regarding the `run_tests` method, it looks more like a private method that doesn't need to be exposed.